### PR TITLE
New pack: HashiCorp Terraform

### PR DESCRIPTION
**Pull request by ilappe**
## Related Issues
fixes:https://jira-dc.paloaltonetworks.com/browse/CIAC-3891

## Description
new pack with one integration

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+      description: 'Link to the Terraform plan data.'
+      type: String
+    - contextPath: Terraform.Plan.links.json-output
+      description: 'Link to the JSON output of the Terraform plan.'
+      type: String
+    - contextPath: Terraform.Plan.links.json-output-redacted
+      description: 'Link to the redacted JSON output of the Terraform plan.'
+      type: String
+    - contextPath: Terraform.Plan.links.json-schema
+      description: 'Link to the JSON schema of the Terraform plan.'
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:320]:
```suggestion
      description: Type of the Terraform plan data.
      type: String
    - contextPath: Terraform.Plan.attributes.has-changes
      description: Flag indicating whether the Terraform plan has changes.
      type: Boolean
    - contextPath: Terraform.Plan.attributes.status-timestamps.started-at
      description: Timestamp indicating when the Terraform plan started.
      type: Date
    - contextPath: Terraform.Plan.attributes.status-timestamps.finished-at
      description: Timestamp indicating when the Terraform plan finished.
      type: Date
    - contextPath: Terraform.Plan.attributes.status-timestamps.agent-queued-at
      description: Timestamp indicating when the Terraform plan was queued for an agent.
      type: Date
    - contextPath: Terraform.Plan.attributes.log-read-url
      description: URL for reading the Terraform plan log.
      type: String
    - contextPath: Terraform.Plan.attributes.resource-additions
      description: Number of resource additions in the Terraform plan.
      type: Number
    - contextPath: Terraform.Plan.attributes.resource-changes
      description: Number of resource changes in the Terraform plan.
      type: Number
    - contextPath: Terraform.Plan.attributes.resource-destructions
      description: Number of resource destructions in the Terraform plan.
      type: Number
    - contextPath: Terraform.Plan.attributes.resource-imports
      description: Number of resource imports in the Terraform plan.
      type: Number
    - contextPath: Terraform.Plan.attributes.structured-run-output-enabled
      description: Flag indicating whether structured run output is enabled in the Terraform plan.
      type: Boolean
    - contextPath: Terraform.Plan.attributes.generated-configuration
      description: Flag indicating whether the Terraform plan has generated configuration.
      type: Boolean
    - contextPath: Terraform.Plan.attributes.actions.is-exportable
      description: Flag indicating whether the Terraform plan is exportable.
      type: Boolean
    - contextPath: Terraform.Plan.attributes.execution-details.mode
      description: Execution mode details for the Terraform plan.
      type: String
    - contextPath: Terraform.Plan.attributes.permissions.can-export
      description: Flag indicating whether the user has permission to export the Terraform plan.
      type: Boolean
    - contextPath: Terraform.Plan.relationships.state-versions.data
      description: Relationship information for state versions associated with the Terraform plan.
      type: Unknown
    - contextPath: Terraform.Plan.relationships.exports.data
      description: Relationship information for exports associated with the Terraform plan.
      type: Unknown
    - contextPath: Terraform.Plan.links.self
      description: Link to the Terraform plan data.
      type: String
    - contextPath: Terraform.Plan.links.json-output
      description: Link to the JSON output of the Terraform plan.
      type: String
    - contextPath: Terraform.Plan.links.json-output-redacted
      description: Link to the redacted JSON output of the Terraform plan.
      type: String
    - contextPath: Terraform.Plan.links.json-schema
      description: Link to the JSON schema of the Terraform plan.
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+      type: String
+    description: Get plan JSON file or the plan meta data.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:322]:
```suggestion
    description: Get the plan JSON file or the plan meta data.
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 3]
```diff
@@ -0,0 +1,627 @@
...
+  - name: terraform-policies-list
+    description: List the policies for an organization or get a specific policy.
+    arguments:
+    - name: organization_name
+      description: The name of the organization.
+    - name: policy_kind
+      description: If specified, restricts results to those with the matching policy kind value. Valid values are sentinel and opa.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:329]:
```suggestion
      description: If specified, restricts results to those with the matching policy kind value. 
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 4]
```diff
@@ -0,0 +1,627 @@
...
+      auto: PREDEFINED
+      predefined:
+      - sentinel
+      - opa
+    - name: policy_name
+      description: If specified, searching the organization's policies by name.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:335]:
```suggestion
      description: If specified, search the organization's policies by name.
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 5]
```diff
@@ -0,0 +1,627 @@
...
+    - name: policy_id
+      description: If specified, get the specific policy.
+    outputs:
+    - contextPath: Terraform.Policy.id
+      description: The policy id.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:340]:
```suggestion
      description: The policy ID.
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 6]
```diff
@@ -0,0 +1,627 @@
...
+      description: 'Previous page number in the pagination of Terraform policies.'
+      type: Unknown
+    - contextPath: Terraform.Policy.meta.pagination.next-page
+      description: 'Next page number in the pagination of Terraform policies.'
+      type: Unknown
+    - contextPath: Terraform.Policy.meta.pagination.total-pages
+      description: 'Total number of pages in the pagination of Terraform policies.'
+      type: Number
+    - contextPath: Terraform.Policy.meta.pagination.total-count
+      description: 'Total count of Terraform policies.'
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:421]:
```suggestion
      description: Name of the Terraform policy.
      type: String
    - contextPath: Terraform.Policy.attributes.description
      description: Description of the Terraform policy.
      type: Unknown
    - contextPath: Terraform.Policy.attributes.enforce.path
      description: Path for enforcing the Terraform policy.
      type: String
    - contextPath: Terraform.Policy.attributes.enforce.mode
      description: Enforcement mode for the Terraform policy.
      type: String
    - contextPath: Terraform.Policy.attributes.policy-set-count
      description: Count of policy sets associated with the Terraform policy.
      type: Number
    - contextPath: Terraform.Policy.attributes.updated-at
      description: Timestamp indicating when the Terraform policy was last updated.
      type: Date
    - contextPath: Terraform.Policy.attributes.kind
      description: Kind of the Terraform policy.
      type: String
    - contextPath: Terraform.Policy.attributes.enforcement-level
      description: Enforcement level for the Terraform policy.
      type: String
    - contextPath: Terraform.Policy.relationships.organization.data.id
      description: Unique identifier for the organization associated with the Terraform policy.
      type: String
    - contextPath: Terraform.Policy.relationships.organization.data.type
      description: Type of the organization associated with the Terraform policy.
      type: String
    - contextPath: Terraform.Policy.relationships.policy-sets.data.id
      description: The IDs of the policy sets that contain this policy.
      type: String
    - contextPath: Terraform.Policy.relationships.policy-sets.data.type
      description: Type of the policy sets associated with the Terraform policy.
      type: String
    - contextPath: Terraform.Policy.links.self
      description: Link to the Terraform policy data.
      type: String
    - contextPath: Terraform.Policy.links.upload
      description: Link for uploading the Terraform policy.
      type: String
    - contextPath: Terraform.Policy.links.download
      description: Link for downloading the Terraform policy.
      type: String
    - contextPath: Terraform.Policy.links.self
      description: Link to the Terraform policy.
      type: String
    - contextPath: Terraform.Policy.links.first
      description: Link to the first page of Terraform policies.
      type: String
    - contextPath: Terraform.Policy.links.prev
      description: Link to the previous page of Terraform policies.
      type: Unknown
    - contextPath: Terraform.Policy.links.next
      description: Link to the next page of Terraform policies.
      type: Unknown
    - contextPath: Terraform.Policy.links.last
      description: Link to the last page of Terraform policies.
      type: String
    - contextPath: Terraform.Policy.meta.pagination.current-page
      description: Current page number in the pagination of Terraform policies.
      type: Number
    - contextPath: Terraform.Policy.meta.pagination.page-size
      description: Number of policies displayed per page in pagination.
      type: Number
    - contextPath: Terraform.Policy.meta.pagination.prev-page
      description: Previous page number in the pagination of Terraform policies.
      type: Unknown
    - contextPath: Terraform.Policy.meta.pagination.next-page
      description: Next page number in the pagination of Terraform policies.
      type: Unknown
    - contextPath: Terraform.Policy.meta.pagination.total-pages
      description: Total number of pages in the pagination of Terraform policies.
      type: Number
    - contextPath: Terraform.Policy.meta.pagination.total-count
      description: Total count of Terraform policies.
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 7]
```diff
@@ -0,0 +1,627 @@
...
+    - name: policy_set_id
+      description: If specified, get the specific policy set.
+    - name: versioned
+      description: Allows filtering policy sets based on whether they are versioned, or use individual policy relationships. A true value returns versioned sets, and a false value returns sets with individual policy relationships. If omitted, all policy sets are returned.
+      auto: PREDEFINED
+      predefined:
+      - "true"
+      - "false"
+    - name: policy_set_kind
+      description: If specified, restricts results to those with the matching policy kind value. Valid values are sentinel and opa.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:438]:
```suggestion
      description: If specified, restricts results to those with the matching policy kind value. 
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 8]
```diff
@@ -0,0 +1,627 @@
...
+      auto: PREDEFINED
+      predefined:
+      - sentinel
+      - opa
+    - name: include
+      description: Enables you to include related resource data. Value must be a comma-separated list containing one or more of projects, workspaces, workspace-exclusions, policies, newest_version, or current_version.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:444]:
```suggestion
      description: Enables you to include related resource data. Value must be a comma-separated list containing one or more projects, workspaces, workspace-exclusions, policies, newest_version, or current_version.
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 9]
```diff
@@ -0,0 +1,627 @@
...
+    - name: policy_set_name
+      description: Allows searching the organization's policy sets by name.
+    - name: page_number
+      description: The page number of the results to return. Default is 1.
+    - name: page_size
+      description: The number of results to return per page. Default is 20, maximum is 100.
+    outputs:
+    - contextPath: Terraform.PolicySet.id
+      description: The policy set id.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:453]:
```suggestion
      description: The policy set ID.
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 10]
```diff
@@ -0,0 +1,627 @@
...
+      description: 'Link to the previous page in the pagination of Terraform policy sets.'
+      type: Unknown
+    - contextPath: Terraform.PolicySet.meta.pagination.next-page
+      description: 'Link to the next page in the pagination of Terraform policy sets.'
+      type: Unknown
+    - contextPath: Terraform.PolicySet.meta.pagination.total-pages
+      description: 'Total number of pages in the pagination of Terraform policy sets.'
+      type: Number
+    - contextPath: Terraform.PolicySet.meta.pagination.total-count
+      description: 'Total number of Terraform policy sets.'
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:549]:
```suggestion
      description: Name of the Terraform policy set.
      type: String
    - contextPath: Terraform.PolicySet.attributes.description
      description: Description of the Terraform policy set.
      type: Unknown
    - contextPath: Terraform.PolicySet.attributes.global
      description: Flag indicating whether the Terraform policy set is global.
      type: Boolean
    - contextPath: Terraform.PolicySet.attributes.workspace-count
      description: Number of workspaces associated with the Terraform policy set.
      type: Number
    - contextPath: Terraform.PolicySet.attributes.project-count
      description: Number of projects associated with the Terraform policy set.
      type: Number
    - contextPath: Terraform.PolicySet.attributes.created-at
      description: Timestamp indicating when the Terraform policy set was created.
      type: Date
    - contextPath: Terraform.PolicySet.attributes.updated-at
      description: Timestamp indicating when the Terraform policy set was last updated.
      type: Date
    - contextPath: Terraform.PolicySet.attributes.kind
      description: Kind of the Terraform policy set.
      type: String
    - contextPath: Terraform.PolicySet.attributes.agent-enabled
      description: Flag indicating whether the Terraform policy set has agents enabled.
      type: Boolean
    - contextPath: Terraform.PolicySet.attributes.policy-count
      description: Number of policies associated with the Terraform policy set.
      type: Number
    - contextPath: Terraform.PolicySet.attributes.versioned
      description: Flag indicating whether the Terraform policy set is versioned.
      type: Boolean
    - contextPath: Terraform.PolicySet.relationships.organization.data.id
      description: ID of the organization associated with the Terraform policy set.
      type: String
    - contextPath: Terraform.PolicySet.relationships.organization.data.type
      description: Type of the organization associated with the Terraform policy set.
      type: String
    - contextPath: Terraform.PolicySet.relationships.policies.data.id
      description: ID of the policies associated with the Terraform policy set.
      type: String
    - contextPath: Terraform.PolicySet.relationships.policies.data.type
      description:  Type of the policies associated with the Terraform policy set.
      type: String
    - contextPath: Terraform.PolicySet.relationships.workspaces.data.id
      description:'ID of the workspaces associated with the Terraform policy set.
      type: String
    - contextPath: Terraform.PolicySet.relationships.workspaces.data.type
      description: Type of the workspaces associated with the Terraform policy set.
      type: String
    - contextPath: Terraform.PolicySet.relationships.projects.data.id
      description: Relationship information for projects associated with the Terraform policy set.
      type: String
    - contextPath: Terraform.PolicySet.relationships.workspace-exclusions.data
      description: Relationship information for workspace exclusions associated with the Terraform policy set.
      type: Unknown
    - contextPath: Terraform.PolicySet.links.self
      description: Link to the Terraform policy set data.
      type: String
    - contextPath: Terraform.PolicySet.links.self
      description: Link to the Terraform policy set.
      type: String
    - contextPath: Terraform.PolicySet.links.first
      description: Link to the first page of Terraform policy sets.
      type: String
    - contextPath: Terraform.PolicySet.links.prev
      description: Link to the previous page of Terraform policy sets.
      type: Unknown
    - contextPath: Terraform.PolicySet.links.next
      description: Link to the next page of Terraform policy sets.
      type: Unknown
    - contextPath: Terraform.PolicySet.links.last
      description: Link to the last page of Terraform policy sets.
      type: String
    - contextPath: Terraform.PolicySet.meta.pagination.current-page
      description: Current page number in the pagination of Terraform policy sets.
      type: Number
    - contextPath: Terraform.PolicySet.meta.pagination.page-size
      description: Number of items per page in the pagination of Terraform policy sets.
      type: Number
    - contextPath: Terraform.PolicySet.meta.pagination.prev-page
      description: Link to the previous page in the pagination of Terraform policy sets.
      type: Unknown
    - contextPath: Terraform.PolicySet.meta.pagination.next-page
      description: Link to the next page in the pagination of Terraform policy sets.
      type: Unknown
    - contextPath: Terraform.PolicySet.meta.pagination.total-pages
      description: Total number of pages in the pagination of Terraform policy sets.
      type: Number
    - contextPath: Terraform.PolicySet.meta.pagination.total-count
      description: Total number of Terraform policy sets.
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 11]
```diff
@@ -0,0 +1,627 @@
...
+      description: 'Flag indicating whether the user has permission to override the Terraform policy check.'
+      type: Boolean
+    - contextPath: Terraform.PolicyCheck.relationships.run.data.id
+      description: 'Unique identifier for the Terraform run associated with the policy check.'
+      type: String
+    - contextPath: Terraform.PolicyCheck.relationships.run.data.type
+      description: 'Type of the Terraform run associated with the policy check.'
+      type: String
+    - contextPath: Terraform.PolicyCheck.links.output
+      description: 'Link to the output of the Terraform policy check.'
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:618]:
```suggestion
      description: Type of the Terraform policy check data.
      type: String
    - contextPath: Terraform.PolicyCheck.attributes.result.result
      description: Overall result of the Terraform policy check.
      type: Boolean
    - contextPath: Terraform.PolicyCheck.attributes.result.passed
      description: Number of policy checks that passed.
      type: Number
    - contextPath: Terraform.PolicyCheck.attributes.result.total-failed
      description: Total number of policy checks that failed.
      type: Number
    - contextPath: Terraform.PolicyCheck.attributes.result.hard-failed
      description: Number of policy checks that resulted in hard failures.
      type: Number
    - contextPath: Terraform.PolicyCheck.attributes.result.soft-failed
      description: Number of policy checks that resulted in soft failures.
      type: Number
    - contextPath: Terraform.PolicyCheck.attributes.result.advisory-failed
      description: Number of policy checks that resulted in advisory failures.
      type: Number
    - contextPath: Terraform.PolicyCheck.attributes.result.duration-ms
      description: Duration of the policy check execution in milliseconds.
      type: Number
    - contextPath: Terraform.PolicyCheck.attributes.result.sentinel
      description: Sentinel-specific result of the policy check.
      type: Unknown
    - contextPath: Terraform.PolicyCheck.attributes.scope
      description: Scope or context of the Terraform policy check.
      type: String
    - contextPath: Terraform.PolicyCheck.attributes.status
      description: Status of the Terraform policy check.
      type: String
    - contextPath: Terraform.PolicyCheck.attributes.status-timestamps.queued-at
      description: Timestamp indicating when the Terraform policy check was queued.
      type: Date
    - contextPath: Terraform.PolicyCheck.attributes.status-timestamps.soft-failed-at
      description: Timestamp indicating when the Terraform policy check encountered a soft failure.
      type: Date
    - contextPath: Terraform.PolicyCheck.attributes.actions.is-overridable
      description: Flag indicating whether the Terraform policy check is overridable.
      type: Boolean
    - contextPath: Terraform.PolicyCheck.attributes.permissions.can-override
      description: Flag indicating whether the user has permission to override the Terraform policy check.
      type: Boolean
    - contextPath: Terraform.PolicyCheck.relationships.run.data.id
      description: Unique identifier for the Terraform run associated with the policy check.
      type: String
    - contextPath: Terraform.PolicyCheck.relationships.run.data.type
      description: Type of the Terraform run associated with the policy check.
      type: String
    - contextPath: Terraform.PolicyCheck.links.output
      description: Link to the output of the Terraform policy check.
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform_description.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
+## HashiCorp Terraform Help
+
+
+**Note:**
+There are 3 different types of tokens - Organization, User, Team
+A customer should use a User or Team token that has admin level access to the workspace.
+for more details see [HashiCorp API Tokens](https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/api-tokens).
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform_description.md:7]:
```suggestion
For details, see [HashiCorp API Tokens](https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/api-tokens).
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 13]
```diff
@@ -0,0 +1,627 @@
...
+  hiddenusername: true
+  type: 9
+  section: Connect
+- display: Default Organization Name
+  name: default_organization_name
+  type: 0
+  required: false
+  section: Connect
+  additionalinfo: There is an option to override with a command input parameter. If not provided, policies commands should require the Org name.
+- display: Default Workspace Id
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:24]:
```suggestion
- display: Default Workspace ID
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform_description.md [hunk 14]
```diff
@@ -0,0 +1,10 @@
...
+
+- *Default Workspace Id* can be taken from *Projects & workspaces* within the selected organization.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform_description.md:9]:
```suggestion
- *Default Workspace ID* can be taken from *Projects & workspaces* within the selected organization.
```

File: Packs/HashiCorpTerraform/README.md [hunk 15]
```diff
@@ -0,0 +1,8 @@
+HashiCorp Terraform is an Infrastructure as Code (IaC) tool designed to streamline and automate the deployment and management of infrastructure across diverse environments.
+
+## What does this pack do?
+- List Runs in a Workspace.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/README.md:4]:
```suggestion
- List runs in a workspace.
```

File: Packs/HashiCorpTerraform/README.md [hunk 16]
```diff
@@ -0,0 +1,8 @@
...
+- Perform actions on run such as: apply, cancel, discard etc.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/README.md:5]:
```suggestion
- Perform actions on a run such as: apply, cancel, discard, etc.
```

File: Packs/HashiCorpTerraform/README.md [hunk 17]
```diff
@@ -0,0 +1,8 @@
...
+- Show a plan.
+- List Policies Policy Sets and Policy Checks.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorpTerraform/README.md:7]:
```suggestion
- List a policy's policy sets and policy checks.
```

**Comment by ShirleyDenkberg** [comment]
@ilappe  Please regenerate the readme.
@DeanArbel @michal-dagan Doc review completed.

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml [hunk 18]
```diff
@@ -0,0 +1,627 @@
...
+      description: 'Replacement addresses associated with the Terraform run.'
+      type: Unknown
+    - contextPath: Terraform.Run.data.attributes.save-plan
+      description: 'Flag indicating whether the Terraform run plan should be saved.'
+      type: Boolean
+    - contextPath: Terraform.Run.data.attributes.variables
+      description: 'Variables associated with the Terraform run.'
+      type: Unknown
+    - contextPath: Terraform.Run.data.relationships.apply
+      description: 'Relationship information for the Terraform run apply.'
```
Comment (ilappe) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.yml:190]:
```suggestion
    - contextPath: Terraform.Run.data.relationships.apply.data.id
      description: The apply ID of the run.
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.py [hunk 19]
```diff
@@ -0,0 +1,356 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+import urllib3
+from requests import Response
+
+urllib3.disable_warnings()
```
Comment (DeanArbel) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.py:6]: This is taken care of in CSP.
```suggestion
```

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.py [hunk 20]
```diff
@@ -0,0 +1,356 @@
...
+CHECK_HR_KEY_TO_RES_KEY = {
+    'Policy check id': 'id',
+    'Result': 'attributes.result',
+    'Status': 'attributes.status',
+    'Scope ': 'attributes.scope'
+}
+
+
+class Client(BaseClient):
+    def __init__(self, url, token, default_organization_name, default_workspace_id, verify, proxy):
```
Comment (DeanArbel) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.py:45]: Please add types to the signature.

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.py [hunk 21]
```diff
@@ -0,0 +1,356 @@
...
+            if policy_set_kind:
+                params['filter[kind]'] = policy_set_kind
+            if include:
+                params['include'] = include
+            if policy_set_name:
+                params['search[name]'] = policy_set_name
+            if page_number:
+                params['page[number]'] = page_number
+            if page_size:
+                params['page[size]'] = page_size
```
Comment (DeanArbel) [inline Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.py:130]: you can also do it via `assign_params()`.
This message is true for other parts of the code, but this is the best example.

**Review by DeanArbel — APPROVED** [review]
Great job, please make sure to run ruff on the python file (precommit hook)

File: Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.py (hunk 21 above)
Comment (ilappe) [reply Packs/HashiCorpTerraform/Integrations/HashiCorpTerraform/HashiCorpTerraform.py:130]: `assign_params` is problematic here as there is special characters and saved words at the keys
for example: `params['filter[versioned]'] = versioned` should be `assign_params(filter[versioned]=versioned)`

**Pull request merged**